pkg/shp/flags/build.go

```diff
@@ -0,0 +1,64 @@
+package flags
+
+import (
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/spf13/pflag"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BuildSpecFlags creates a BuildSpec instance based on command-line flags.
+func BuildSpecFlags(flags *pflag.FlagSet) *buildv1alpha1.BuildSpec {
```

Comment:
Style nit - make the function name a bit more descriptive:

```suggestion
// BuildSpecFromFlags creates a BuildSpec instance based on command-line flags.
func BuildSpecFromFlags(flags *pflag.FlagSet) *buildv1alpha1.BuildSpec {
```